### DISCUSSION TITLE
Error When Using V4.5.1 and Carbon V3

### DISCUSSION THREAD

**Discussion by phillipbcoyle** [spatie/laravel-honeypot#130] [General]
This package does not appear to be compatible with Carbon V3.

When using the Honeypot package V4.5.1 and Carbon V3, I receive the following error on any website page that has this package being used in a Livewire component.

```Carbon\Carbon::rawAddUnit(): Argument #3 ($value) must be of type int|float, string given, called in vendor/nesbot/carbon/src/Carbon/Traits/Units.php on line 356```

If I downgrade Carbon to V2, or if I limit the Honeypot package to V4.5.0 (the last version requiring Carbon V2 only), then it works.

**Comment by phillipbcoyle** [comment]
Fixed in V4.5.2

**Reply by yanikkumar** [reply]
It's still not fixed yet (completely). 
It creates the same issue with version v4.5.2 and Laravel version v11.21.0
I don't want to downgrade the Laravel so I did downgraded the honeypot-package to v4.5.0 and Carbon was downgraded as well. Now things are working normally again but 
What is the real issue can anyone explain and how to fix it permanently.

**Reply by yanikkumar** [reply]
I fixed this thing by just casting the `amount_of_seconds` in `config\honepot.php` 

`'amount_of_seconds' => (int) env('HONEYPOT_SECONDS', 3),`

Since .env values are all strings, config is being set as a string rather than an int, so I change the type.
I hope this will help someone, but still, this is not the exact fix because we are modifying the config file.
